### PR TITLE
LL-3997 hard reset 

### PR DESCRIPTION
**Pull request by machard**
https://ledgerhq.atlassian.net/browse/LL-3997

### CODE REVIEW

File: src/renderer/screens/settings/sections/Help/ResetButton.js [hunk 1]
```diff
@@ -3,17 +3,20 @@
...
 import ConfirmModal from "~/renderer/modals/ConfirmModal";
 import ResetFallbackModal from "~/renderer/modals/ResetFallbackModal";
-import { SyncSkipUnderPriority } from "@ledgerhq/live-common/lib/bridge/react";
 import Box from "~/renderer/components/Box";
 import Button from "~/renderer/components/Button";
 import IconTriangleWarning from "~/renderer/icons/TriangleWarning";
 import type { ThemedComponent } from "~/renderer/styles/StyleProvider";
 import { useActionModal } from "./logic";
 
+let reset;
```
Comment (gre) [inline src/renderer/screens/settings/sections/Help/ResetButton.js:18]: i'm not really sure the global variable pattern is a good idea. This button could be used at different places too.

File: src/renderer/screens/settings/sections/Help/ResetButton.js [hunk 2]
```diff
@@ -22,17 +25,28 @@ export default function ResetButton() {
...
 
-  const onConfirm = useCallback(async () => {
-    if (pending) return;
+  const globalSyncState = useGlobalSyncState();
+  const cvPolling = useCountervaluesPolling();
+  const isPending = cvPolling.pending || globalSyncState.pending;
+
+  reset = async () => {
+    if (isPending) {
+      return setTimeout(reset, 1000);
```
Comment (gre) [inline src/renderer/screens/settings/sections/Help/ResetButton.js:34]: setTimeout do not return a promise so i'm not sure what is going on here.

wouldn't it be simpler to have a "isPending()" predicate that you call in a

```
while(isPending()) {
  await delay(1000);
}
```

**Review by gre — CHANGES_REQUESTED** [review]
i'm not fan of the current implementation, but also I don't think it's a good idea to implement it this way.

a sync and a fetch countervalues can possibly takes A LOT of time. to me the reset action should be near instant, but should cancel all ongoing actions that are happening. (like kill the thread, stop from receiving any REDUX events so we don't "race condition" on the "save again" situation)

File: src/renderer/screens/settings/sections/Help/ResetButton.js (hunk 1 above)
Comment (machard) [reply src/renderer/screens/settings/sections/Help/ResetButton.js:18]: yes indeed i should have put it in a ref

File: src/renderer/screens/settings/sections/Help/ResetButton.js (hunk 2 above)
Comment (machard) [reply src/renderer/screens/settings/sections/Help/ResetButton.js:34]: i am not retuning a promise, i am just waiting to recall the function.
indeed i did not think to do it this way. (need to be careful with the isPending to be sure it uses an updated version of isPending in the logic though).

**Comment by machard** [comment]
(i removed the wiping of the countervalues, but I assume since we clean everything at the beginning, the countervalue will be initiated with the initialState)

**Review by gre — APPROVED** [review]

**Review by nabil-brn — APPROVED** [review]
Tested with 70 accounts _app.json_.
`,"accounts":[]}}`

---
Windows
fd8058e23 (HEAD -> pr/machard/3381, machard/LL-3997) go the other way around

**Pull request merged**